src/features/joinForms/panes/JoinFormPane.tsx

```diff
@@ -0,0 +1,132 @@
+import { FC } from 'react';
+import {
+  Autocomplete,
+  Box,
+  IconButton,
+  TextField,
+  useTheme,
+} from '@mui/material';
+
+import { Delete } from '@mui/icons-material';
+import globalMessageIds from 'core/i18n/globalMessageIds';
+import messageIds from '../l10n/messageIds';
+import { NATIVE_PERSON_FIELDS } from 'features/views/components/types';
+import PaneHeader from 'utils/panes/PaneHeader';
+import useCustomFields from 'features/profile/hooks/useCustomFields';
+import useJoinForm from '../hooks/useJoinForm';
+import useJoinFormMutations from '../hooks/useJoinFormMutations';
+import { useMessages } from 'core/i18n';
+
+type Props = {
+  formId: number;
+  orgId: number;
+};
+
+const JoinFormPane: FC<Props> = ({ orgId, formId }) => {
+  const { data } = useJoinForm(orgId, formId);
+  const { updateForm } = useJoinFormMutations(orgId, formId);
+  const messages = useMessages(messageIds);
+  const globalMessages = useMessages(globalMessageIds);
+  const customFields = useCustomFields(orgId);
+  const theme = useTheme();
+
+  if (!data) {
+    return null;
+  }
+
+  function slugToLabel(slug: string): string {
+    const isNativeField = Object.values(NATIVE_PERSON_FIELDS).some(
+      (nativeSlug) => nativeSlug == slug
+    );
+    if (isNativeField) {
+      const typedSlug = slug as NATIVE_PERSON_FIELDS;
+      return globalMessages.personFields[typedSlug]();
+    } else {
+      const field = customFields.data?.find((field) => field.slug == slug);
+      return field?.title ?? '';
+    }
+  }
+
+  return (
+    <>
+      <PaneHeader title={messages.formPane.title()} />
+      <Box mb={2}>
+        <TextField
+          defaultValue={data.title}
+          fullWidth
+          label={messages.formPane.labels.title()}
+          onChange={(evt) => updateForm({ title: evt.target.value })}
+        />
+      </Box>
+      <Box mb={1}>
+        <TextField
+          defaultValue={data.description}
+          fullWidth
+          label={messages.formPane.labels.description()}
+          onChange={(evt) => updateForm({ description: evt.target.value })}
+        />
+      </Box>
+      <Box mb={1}>
+        <Autocomplete
+          fullWidth
+          getOptionDisabled={(option) => data.fields.includes(option)}
+          getOptionLabel={slugToLabel}
+          onChange={(ev, value) => {
+            if (value) {
+              updateForm({ fields: [...data.fields, value] });
+            }
+          }}
+          options={[
+            ...Object.values(NATIVE_PERSON_FIELDS),
+            ...(customFields.data?.map((field) => field.slug) ?? []),
+          ]}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              placeholder={messages.formPane.labels.addField()}
+            />
+          )}
+          value={null}
```

Comment:
The purpose of the `null` value is to make sure that even if the user selects something, it does not stay selected. That isn't working great though, because it still fills the input with the name and as such filters the options down.

But I do believe that if the value is not `null`, it will select whatever the user clicks (because the component will not be "controlled").